src/azure-cli-core/azure/cli/core/tests/test_keys.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import unittest
+import tempfile
+import paramiko
+import io
+import os
+import shutil
+from knack.util import CLIError
+
+from azure.cli.core.keys import generate_ssh_keys
+
+
+class TestGenerateSSHKeys(unittest.TestCase):
+
+    def setUp(self):
+        # set up temporary directory to be used for temp files.
+        self._tempdirName = tempfile.mkdtemp(prefix="key_tmp_")
+
+        self.key = paramiko.RSAKey.generate(2048)
+        keyOutput = io.StringIO()
+        self.key.write_private_key(keyOutput)
+
+        self.private_key = keyOutput.getvalue()
+        self.public_key = '{} {}'.format(self.key.get_name(), self.key.get_base64())
+
+    def tearDown(self):
+        # delete temporary directory to be used for temp files.
+        shutil.rmtree(self._tempdirName)
+
+    def test_public_key_file_exists(self):
+        # Create public key file
+        public_key_path = self._create_new_temp_key_file(self.public_key, suffix=".pub")
+
+        # Create private key file
+        private_key_path = self._create_new_temp_key_file(self.private_key)
+
+        # Call generate_ssh_keys and assert that returned public key same as original
+        new_public_key = generate_ssh_keys("", public_key_path)
+        self.assertEqual(self.public_key, new_public_key)
+
+        # Check that private and public key file contents unchanged.
+        with open(public_key_path, 'r') as f:
+            new_public_key = f.read()
+            self.assertEqual(self.public_key, new_public_key)
+
+        with open(private_key_path, 'r') as f:
+            new_private_key = f.read()
+            self.assertEqual(self.private_key, new_private_key)
+
+    def test_public_key_file_exists_no_permissions(self):
```

Comment:
again, we don't need this test

Reply:
If you would like me to remove this test, I can. Although @troydai asked me to test different cornercases that come to mind.